ci/test-clang.sh

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env bash
+
+set -eu
+
+trap 'rm -fr "$toolchain"' EXIT
+
+# Set up workspace
+rootdir=${GITHUB_WORKSPACE:-$(dirname "$(dirname "$(realpath "$0")")")}
+toolchain=$rootdir/toolchain
+
+# Pull toolchain out of container
+echo "[+] Downloading toolchain from container"
+docker create --name llvm-project ghcr.io/clangbuiltlinux/llvm-project:stage2
+mkdir -p "$toolchain"/{bin,include,lib}
+docker cp llvm-project:/usr/local/bin/. "$toolchain"/bin
+docker cp llvm-project:/usr/local/include/. "$toolchain"/include
+docker cp llvm-project:/usr/local/lib/. "$toolchain"/lib
+echo "[+] Cleaning up container"
+docker rm llvm-project
+
+# Test toolchain in Docker images so that the environment is consistent,
+# regardless of runners.
+docker_images=(
+    docker.io/fedora:latest
```

Comment:
This should work in moar env's, right? RIGHT?!? Let's add more else file an issue/todo.

Reply:
oh, most other distros wont have `source /usr/lib/os-release`?

Reply:
I am having trouble parsing your first comment ("moar env"?).

Any distribution that uses systemd should have `/usr/lib/os-release`, it is a part of their base specification: https://www.freedesktop.org/software/systemd/man/os-release.html

That page says `/etc/os-release` is preferred but most distributions have it as a symlink to `/usr/lib/os-release`.

Reply:
"Let's add additional `docker_images` to test in, like alpine, debian, etc, or a todo issue in the issue tracker"

Reply:
Aha, yes, that is the idea. I have already wired up Arch Linux in a separate change:

https://github.com/nathanchance/containers/commit/13001a2067cc88283252272c5e9a84f1f59ce98e

https://github.com/nathanchance/containers/runs/6601565398?check_suite_focus=true

I'll file an issue to look into other distributions as well.

Reply:
I opened https://github.com/ClangBuiltLinux/containers/issues/14. I pushed the Arch Linux support as part of this PR in 5f2bbadc55a717f16a372efb98b6f5b772fd7553, as I don't really see a point in doing that in a follow up since it is tested and ready to go now.